babel/src/test/resources/sql/big-query.iq

```diff
@@ -131,7 +131,7 @@ D
 
 select date(cast(null as varchar(10))) as d;
 D
-null
```

Comment:
Calcite will parse DATE(NULL) to 1970-01-01

Reply:
What exactly is causing this? Will this happen for Bodo? If so, this needs to be addressed. If not, we can probably let it slide.

Reply:
This won't happen for Bodo because it will calls our to_date python kernel. I ran this query is [this branch](https://github.com/Bodo-inc/Bodo/pull/5129) and here is the result:
```
SQL query:
select date(cast(null as varchar(10)))

Unoptimized plan:
LogicalProject(EXPR$0=[DATE(null:VARCHAR(10) CHARACTER SET "UTF-8")])
  LogicalValues(tuples=[[{ 0 }]])


Optimized plan:
LogicalProject(EXPR$0=[DATE(null:VARCHAR(10) CHARACTER SET "UTF-8")])
  LogicalValues(tuples=[[{ 0 }]])


Generated code:
  df2 = pd.DataFrame({"ZERO": bodo.utils.conversion.coerce_scalar_to_array(np.int32(0), 1, global_1), }, index=bodo.hiframes.pd_index_ext.init_range_index(0, 1, 1, None))
  index = pd.RangeIndex(0, len(df2), 1)
  S3 = bodo.utils.conversion.coerce_scalar_to_array(bodo.libs.bodosql_array_kernels.date(None, None), len(df2), global_2)
  T4 = bodo.hiframes.table.logical_table_to_table(bodo.hiframes.pd_dataframe_ext.get_dataframe_all_data(df2), (S3, ), global_3, df2.shape[1])
  df1 = bodo.hiframes.pd_dataframe_ext.init_dataframe((T4,), index, global_4)
  return df1

```
And `bodo.libs.bodosql_array_kernels.date(None, None)` returns None.

Reply:
Ok, I'm ok to approve this then.